core/src/main/kotlin/endpoints/AgentManager.kt

```diff
@@ -118,6 +119,7 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
             session.updateSessionHeader(adminData.settings.sessionIdHeaderName)
             currentInfo
         } else {
+            logger.debug { "agent($id, $buildVersion): attaching to new build..." }
```

Comment:
This is not the case for some new build. Only if isNewBuild is true this is a new build.